doc/sphinx/architecture.md

```diff
@@ -0,0 +1,131 @@
+- The classes that the `Server` relies on for its runtime can be found organized in separate folders in this directory.
+
+## `/src/server/compositor`
+- The `Compositor` is responsible for collecting the set of renderables and sending them off to the outputs to be displayed.
+
+## `/src/server/input`
+- This folder deals with everything input (e.g. keyboard presses, mouse clicks)
+- The `InputManager` provides access to the specific input platform that is running (e.g. X, Wayland, or libinput).
+- The `InputDispatcher` is responsible for taking events bubbled up from the platform and sending them off to the appropriate listeners
+
+## `/src/server/frontend_wayland`
+- The *Mir* implementation of the wayland protocol.
+- The `WaylandConnector` class is the glue between the compositor and the specific details of the wayland protocol.
+
+## `/src/server/frontend_xwayland`
+- The *Mir* implementation of the xwayland protocol.
+- The `XWaylandConnector` class is the glue between the compositor and the specific details of the xwayland protocol.
+- This protocol talks to the `xwayland` server, which is spawned as a subprocess.
+
+## `/src/server/shell`
+- The `Shell` is responsible for managing the surfaces that the compositor wants to show.
+- It provides an interface to create, update, and remove these surfaces from the display.
+- As an example, the `WaylandConnector` might ask the `Shell` to create a new surface for it.
+
+## `/src/server/scene`
+- The `Scene` provides an interface for the `Compositor` to access the list of renderable items
+  - These renderables are derived from the surfaces that were added to the `Shell`.
+  - You can think of the `Scene` as you would think of a "scene graph" in a 3D game engine.
+- Unlike the `Shell`, the `Scene` knows a lot about the Z-order of the surfaces. For this reason, it is also responsible for things like locking the display or sending surfaces to the back of the Z-order.
+
+## `/src/server/graphics`
+- A graphics abstraction layer sitting between the compositor and the specific graphics platform that it is using to render.
+
+## `/src/server/console`
+- Handles `logind` and virtual-terminal related tasks for the compositor
+
+## `/src/server/report`
+- Logging and other facilities to collect data about the compositor
+
+## `/src/platforms` (with an "s"!)
+- Provides both input and rendering platforms that the compositor can use depending on the environment.
+- For the rendering platforms, we have:
+  - *GBM/KMS*: the platform that you would expect if you were running a *Mir*-based compositor as your daily-driver
+  - *X11*: a useful platform for testing locally
+  - *Wayland*: another useful platform for testing locally
+  - *EGLStreams/KMS*: like the *GBM/KMS* platform, but for Nvidia 
+- For the input platforms, we have:
+  - evdev
+  - X11
+  - Wayland
+
+## `/src/platform` (no "s")
+- Graphics and input code that is shared between platforms
```

Comment:
Comment: In principle, we aim to make this a public interface for third party "platforms". As Chris has been reworking these APIs, they are currently private